CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Screen API Coordinator/GiniScreenAPICoordinator.swift

```diff
@@ -196,35 +196,49 @@ extension GiniScreenAPICoordinator {
     @objc func back() {
         switch screenAPINavigationController.topViewController {
         case is CameraViewController:
-            trackingDelegate?.onCameraScreenEvent(event: Event(type: .exit))
-            // TODO: rethink this when you add Camera access screen events!
-            // is also accessed from that screen
-            // do we need a different name for the screen in that case?
-            AnalyticsManager.track(event: .closeTapped, screenName: .camera)
-
-            if pages.type == .qrcode {
-                screenAPINavigationController.dismiss(animated: true)
-            }
-
-            if pages.count > 0 {
-                if screenAPINavigationController.viewControllers.count > 1 {
-                    screenAPINavigationController.popViewController(animated: true)
-                } else {
-                    screenAPINavigationController.dismiss(animated: true)
-                }
-            } else {
-                closeScreenApi()
-            }
+            handleCameraViewControllerBack()
         case is AnalysisViewController:
-            trackingDelegate?.onAnalysisScreenEvent(event: Event(type: .cancel))
-            AnalyticsManager.track(event: .closeTapped, screenName: .analysis)
-            screenAPINavigationController.dismiss(animated: true)
+            handleAnalysisViewControllerBack()
         default:
-            if screenAPINavigationController.viewControllers.count > 1 {
-                screenAPINavigationController.popViewController(animated: true)
-            } else {
-                screenAPINavigationController.dismiss(animated: true)
-            }
+            handleDefaultBack()
+        }
+    }
+
+    private func handleCameraViewControllerBack() {
+        trackingDelegate?.onCameraScreenEvent(event: Event(type: .exit))
+        AnalyticsManager.track(event: .closeTapped, screenName: getScreenName())
+        if pages.type == .qrcode {
+            screenAPINavigationController.dismiss(animated: true)
+        }
```

Comment:
[Suggestion]: I think `return` is missing here in line 212, we don't need to go further in the code.

Reply:
I don't think is needed:

Reply:
@zladzeyka I might need some extra clarifications for this. 
Can we please have a short call whenever you have time? Thank you!

Reply:
I'm on this right now. Thanks for your input. I also agree with using `guard` whenever is possible instead of if.

Reply:
As a small remark, `screenAPINavigationController.dismiss(animated: true)` we dismiss our entire SDK.